src/FreeAge/client/unit.hpp

```diff
@@ -116,17 +116,24 @@ class ClientUnit : public ClientObject {
   
   Texture& GetTexture(bool shadow);
   
-  inline const QPointF& GetMapCoord() const { return mapCoord; }
+  inline const QPointF& GetMapCoord() const { 
+    CHECK_EQ(IsGarrisoned(), false);
```

Comment:
Currently the position of a unit that is garrisoned should not be used. I originally was setting it to the center of the building that is inside. Then the problem was if its inside a unit (transport ship, ram) it would have to be constantly updated. For now if the position is needed (eg. click "go to idle villager" and the villager is inside a building) it could be calculated by finding the center of the `unit->garrisonedObjectId`

Reply:
If GetMapCoord() had access to the object in which the unit is garrisoned in, it could even obtain the position of the garrison-target itself and return that.

In any case, I would suggest to turn the `CHECK_EQ(IsGarrisoned(), false);` into something like:
```c++
if (IsGarrisoned()) {
  LOG(ERROR) << "This must not be called on garrisoned units.";
}
```
This way, the game will probably not crash immediately if something erroneously calls that once.

Reply:
> If GetMapCoord() had access to the object in which the unit is garrisoned in, it could even obtain the position of the garrison-target itself and return that.

I believe you mean storing a `ClientObject*` instead of a ID. My original idea was to use ID in order to have the same code in case a `Object` will become a thing (a super class of ClientObject and ServerObject), but now I see that in that case a pointer of type `Object` could replace the `ClientObject` without any problem.

Should I just replace the ID with pointers here and in `ClientObject -> std::vector<u32> garrisonedUnits`?

Plus: the ID storage thing would not be an issue

Reply:
> I believe you mean storing a ClientObject* instead of a ID.

There would also be the possibility of getting hold of a game `Map` pointer somehow and then obtaining the object via its id from the map.

> Should I just replace the ID with pointers here and in ClientObject -> std::vector<u32> garrisonedUnits?

That seems like a possibility, yes. I currently don't have a good overview of all possible implications, but if no issues show up during implementation, then that seems like it might be convenient.

Reply:
I changed the code to use pointers, there a place where the IDs would help but I will leave it for now with `NOTE: ... #ids` comment. It can revisited in the future